packages/test-sync/src/androidAndroidTest/kotlin/io/realm/kotlin/test/mongodb/shared/ProgressListenerTests.kt

```diff
@@ -0,0 +1,292 @@
+                // Ensure that all data is already synced
+                assertTrue { syncSession.uploadAllLocalChanges() }
+                assertTrue { syncSession.downloadAllServerChanges() }
+                // Ensure that progress listeners are triggered at least one time even though there
+                // is no data
+                syncSession.progress(Direction.DOWNLOAD, ProgressMode.CURRENT_CHANGES).first()
+                syncSession.progress(Direction.UPLOAD, ProgressMode.CURRENT_CHANGES).first()
+                syncSession.progress(Direction.DOWNLOAD, ProgressMode.INDEFINITELY).first()
+                syncSession.progress(Direction.UPLOAD, ProgressMode.INDEFINITELY).first()
+            }
+        }.close()
+    }
+
+    @Test
+    fun throwsOnFlexibleSync() = runBlocking {
+        val app = TestApp(TEST_APP_FLEX)
+        val user = app.createUserAndLogIn()
+        val configuration: SyncConfiguration = SyncConfiguration.create(user, schema)
+        Realm.open(configuration).apply {
+            assertFailsWithMessage<UnsupportedOperationException>(
+                "Progress listeners are not support for Flexible Sync"
+            ) {
+                syncSession.progress(Direction.DOWNLOAD, ProgressMode.CURRENT_CHANGES)
+            }
+        }.close()
+    }
+
+    private suspend fun Realm.writeSampleData(count: Int, idOffset: Int = 0, timeout: Duration? = null) {
+        write {
+            for (i in idOffset until count + idOffset) {
+                copyToRealm(ProgressTestObject().apply { id = "Object $i" })
+            }
+        }
+        timeout?.let {
+            assertTrue { syncSession.uploadAllLocalChanges(timeout) }
+        }
+    }
+
+    // Operator that will return a flow that emits an incresing integer on each completion event
+    private fun Flow<Progress>.completionCounter(): Flow<Int> =
+        filter { it.isTransferComplete }
+            .distinctUntilChanged()
+            // Increment completed count if we are done transferring and the amount of bytes has
+            // increased
+            .scan(0UL to 0) { (bytes, completed), progress ->
+                if (progress.isTransferComplete && progress.transferableBytes > bytes) {
+                    (progress.transferredBytes to completed + 1)
+                } else {
+                    (bytes to completed)
+                }
+            }
+            .drop(1)
+            .map { (_, completed) -> completed }
+
+    private fun createSyncConfig(
+        user: User,
+        partitionValue: String = getTestPartitionValue()
+    ): SyncConfiguration {
+        return SyncConfiguration.Builder(user, partitionValue, schema)
+            .build()
+    }
+
+    private fun getTestPartitionValue(): String {
+        if (!this::partitionValue.isInitialized) {
+            fail("Test not setup correctly. Partition value is missing")
+        }
+        return partitionValue
+    }
+
+    class ProgressTestObject : RealmObject {
```

Comment:
Nitpick: Do we really need this or cannot we reuse Sample or AllTypes instead?

Reply:
There is a heavy amount of operations with creating objects of `SyncObjectWithAllTypes` compared to what we are actually testing, but 🤷  ... I just reduced the number of objects that we are creating to prevent hitting the timeout limits.